saleor/checkout/complete_checkout.py

```diff
@@ -757,13 +751,352 @@ def complete_checkout(
+        add_voucher_usage_by_customer(voucher, customer_email)
+
+    if voucher.usage_limit:
+        increase_voucher_usage(voucher)
+
+
+def _create_order_lines_from_checkout_lines(
+    checkout_info: CheckoutInfo,
+    lines: List[CheckoutLineInfo],
+    discounts: List["DiscountInfo"],
+    manager: "PluginsManager",
+    order_pk: int,
+    reservation_enabled: bool,
+    taxes_included_in_prices: bool,
+) -> List[OrderLineInfo]:
+    order_lines_info = _create_lines_for_order(
+        manager,
+        checkout_info,
+        lines,
+        discounts,
+        reservation_enabled,
+        taxes_included_in_prices,
+    )
+    order_lines = []
+    for line_info in order_lines_info:
+        line = line_info.line
+        line.order_id = order_pk
+        order_lines.append(line)
+
+    OrderLine.objects.bulk_create(order_lines)
+    return list(order_lines_info)
+
+
+def _handle_allocations_of_order_lines(
+    checkout_info: CheckoutInfo,
+    checkout_lines: List[CheckoutLineInfo],
+    order_lines_info: List[OrderLineInfo],
+    manager: "PluginsManager",
+    reservation_enabled: bool,
+):
+    country_code = checkout_info.get_country()
+    additional_warehouse_lookup = (
+        checkout_info.delivery_method_info.get_warehouse_filter_lookup()
+    )
+    allocate_stocks(
+        order_lines_info,
+        country_code,
+        checkout_info.channel.slug,
+        manager,
+        additional_warehouse_lookup,
+        check_reservations=reservation_enabled,
+        checkout_lines=[line.line for line in checkout_lines],
+    )
+    allocate_preorders(
+        order_lines_info,
+        checkout_info.channel.slug,
+        check_reservations=reservation_enabled,
+        checkout_lines=[line.line for line in checkout_lines],
+    )
+
+
+def _handle_checkout_discount(
+    order: "Order", checkout: "Checkout", voucher: Optional["Voucher"]
+):
+    if checkout.discount:
+        # store voucher as a fixed value as it this the simplest solution for now.
+        # This will be solved when we refactor the voucher logic to use .discounts
+        # relations
+
+        # When we have a voucher for specific products we track it directly in the
+        # Orderline. Voucher with 'apply_once_per_order' is handled in the same way
+        # as we apply it only for single quantity of the cheapest line.
+        if not voucher or (
+            voucher.type != VoucherType.SPECIFIC_PRODUCT
+            and not voucher.apply_once_per_order
+        ):
+            order.discounts.create(
+                type=OrderDiscountType.VOUCHER,
+                value_type=DiscountValueType.FIXED,
+                value=checkout.discount.amount,
+                name=checkout.discount_name,
+                translated_name=checkout.translated_discount_name,
+                currency=checkout.currency,
+                amount_value=checkout.discount_amount,
+            )
+
+
+def _post_create_order_actions(
```

Comment:
Could we perhaps reuse this for order creation in `checkoutComplete`?

Reply:
I didn't want to "touch" checkoutComplete logic for some reasons:
- `orderFromCheckoutCreate` is a preview feature, so the workflow can be changed or even fully deleted. I didn't want to have a lot of common dependencies that could be changed or can be problematic in future changes. 
- If we decide to remove the preview label for `orderFromCheckoutCreate`,  `checkoutComplete` will be deprecated and moved to the `deprecated` directory with all logic.